qiskit_ibm_runtime/utils/noise_learner_result.py

```diff
@@ -0,0 +1,109 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""NoiseLearner result class"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Iterator, Sequence
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.quantum_info import PauliList
+from qiskit.primitives import EstimatorResult
+
+
+@dataclass(frozen=True)
+class NoiseLearnerDatum:
+    """A container for a single noise learner datum.
+
+    Args:
+        circuit: A circuit whose noise has been learnt.
+        qubits: The labels of the qubits in the given circuit.
+        generators: The list of generators for the Pauli errors.
+        rates: The rates for the given Pauli error generators.
+
+    Raises:
+        ValueError: If ``circuit``, ``qubits``, and ``generators`` have mismatching number of
+            qubits.
+        ValueError: If ``generators`` and ``rates`` have different lengths.
+    """
+
+    circuit: QuantumCircuit
+    qubits: Sequence[int]
+    generators: PauliList
+    rates: Sequence[float]
+
+    def __post_init__(self) -> None:
+        if len({self.circuit.num_qubits, len(self.qubits), self.generators.num_qubits}) != 1:
+            raise ValueError("Mistmatching numbers of qubits.")
+        if len(self.generators) != len(self.rates):
+            msg = f"``generators`` has length {len(self.generators)}, but "
+            msg += f"``rates`` has length {len(self.rates)}."
+            raise ValueError(msg)
+
+
+class NoiseLearnerResult:
+    """A container for the results of a noise learner experiment."""
+
+    def __init__(self, data: Sequence[NoiseLearnerDatum], metadata: dict[str, Any] | None = None):
+        """
+        Args:
+            data: The data of a noise learner experiment.
+            metadata: Metadata that is common to all pub results; metadata specific to particular
+                pubs should be placed in their metadata fields. Keys are expected to be strings.
+        """
+        self._data = data
+        self._metadata = metadata or {}
+
+    @classmethod
+    def from_estimator_result(cls, result: EstimatorResult) -> NoiseLearnerResult:
```

Comment:
Is the noise model always returned if ZNE is specified? Or does it require some option?

Reply:
I think that the noise is only specified if the estimator did noise learning. So for example, if the estimator did ZNE with gate folding, it won't be there

Reply:
I think that's worth pointing out in the docstring - i.e. not all `EstimatorResult` will contain noise model, and that an empty `NoiseLearnerResult` is returned if it doesn't. 

Also, do we want to return an empty `NoiseLearnerResult` or just raise?

Reply:
Honestly, when I added this function I thought I'd need it, but I ended up not using it anywhere else. So:
> do we want to return an empty NoiseLearnerResult or just raise?
I don't know right now, I'd like to see a usecase before I decide. If you are okay with it, I'll remove it for now

I think that there are a few more methods that may turn out to be useful here.`from_estimator_result` is one, but also `to_noise_model` to get a `NoiseModel` object that can be used in a simulator, plotting methods..
I will start working on it as soon as I have time :)